ros/src/waypoint_follower/src/pure_pursuit_core.cpp

```diff
@@ -271,6 +288,43 @@ geometry_msgs::Twist PurePursuit::calcTwist(double curvature, double cmd_velocit
   return twist;
 }
 
+// edufford start
+// Search for closest waypt from current pose
+void PurePursuit::getClosestWaypoint() {
+  int path_size = static_cast<int>(current_waypoints_.getSize());
+  int closest_wp = path_size - 1;
+
+  // If waypoints are not given, do nothing
+  if (path_size == 0) {
```

Comment:
Perhaps you might want `if (path_size < 1)` so that we're also protected if `current_waypoints` has a zero size?

Reply:
If current_waypoints has zero size, path_size should be 0 so this should be ok.  I think it was confusion because I initialized `closest_wp = path_size - 1` above, but it should be moved to below this catch so I moved it down.

Reply:
Ah, my mistake! I was confused.